src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1260,6 +1260,65 @@ internal virtual FieldSymbol FixedElementField
         /// <returns>True if this is an interface type.</returns>
         internal abstract bool IsInterface { get; }
 
+        /// <summary>
+        /// Verify if the given type can be used to back a tuple type and return cardinality of that tuple type in <paramref name="tupleCardinality"/>. 
+        /// </summary>
+        /// <param name="tupleCardinality">If method returns true, contains cardinality of the compatible tuple type.</param>
+        /// <returns></returns>
+        public override bool IsTupleCompatible(out int tupleCardinality)
```

Comment:
I thin we should use cardinality (arity is "reserved" for generics and length is too general). In any case, this is not a public API, the type is internal.

---
In reply to: [61791522](https://github.com/dotnet/roslyn/pull/11006#discussion_r61791522) [](ancestors = 61791522)